cscs-checks/libraries/math/trilinos_compile_run.py

```diff
@@ -1,40 +1,42 @@
 import os
-import reframe.utility.sanity as sn
-
-from reframe.core.pipeline import RegressionTest
 
+import reframe as rfm
+import reframe.utility.sanity as sn
 
-class TrilinosTest(RegressionTest):
-    def __init__(self, linkage, **kwargs):
-        super().__init__('trilinos_compile_run_%s' % linkage,
-                         os.path.dirname(__file__), **kwargs)
-
-        self.flags = ' -DHAVE_MPI -DEPETRA_MPI -lparmetis -%s ' % linkage
-        self.prgenv_flags = {
-            'PrgEnv-cray': ' -homp -hstd=c++11 -hmsglevel_4 ',
-            'PrgEnv-gnu': ' -fopenmp -std=c++11 -w -fpermissive ',
-            'PrgEnv-intel': ' -qopenmp -w -std=c++11 ',
-            'PrgEnv-pgi': ' -mp -w '
-        }
-        self.descr = 'Trilinos ' + linkage.capitalize()
-        self.sourcepath = 'example_AmesosFactory_HB.cpp'
-        #self.sourcepath = 'trilinos_compile_run.cpp'
-        self.input_file = os.path.join(self.current_system.resourcesdir,
-                                       'Trilinos', 'trilinos_compile_run.rua')
 
-        self.executable_opts = self.input_file.split()
+@rfm.parameterized_test(['static'], ['dynamic'])
+class TrilinosTest(rfm.RegressionTest):
+    def __init__(self, linkage):
+        super().__init__()
         self.valid_systems = ['daint:gpu', 'daint:mc',
                               'dom:gpu', 'dom:mc']
 
-        # Removed CRAY env in dynamic because of CrayBug/809265
-        if linkage == 'dynamic':
+        # NOTE: PrgEnv-cray in dynamic does not work because of CrayBug/809265
+        # PrgEnv-cray seg faults for cray-trilinos/12.12.1.1
```

Comment:
I don't think this note has to be in this test, since you are not doing anything special for it, do you? If the bug is fixed, the test will pass automatically. I suggest putting this note in the PR description instead.